viadot/sources/databricks.py

```diff
@@ -0,0 +1,436 @@
+                self._handle_if_empty(if_empty=if_empty)
+        else:
+            df = pd.DataFrame()
+        return df
+
+    def _pandas_df_to_spark_df(self, df: pd.DataFrame) -> spark.DataFrame:
+        """
+        Convert a Pandas DataFrame to a Spark DataFrame.
+
+        Args:
+            df (pd.DataFrame): The Pandas DataFrame to be converted to a Spark DataFrame.
+
+        Example:
+        ```python
+        from viadot.sources import Databricks
+        import pandas as pd
+        databricks = Databricks()
+        list = [{"id":"1", "name":"Joe"}]
+        df = pd.DataFrame(list)
+        spark_df = databricks._pandas_df_to_spark_df(df)
+        ```
+        Returns:
+            spark.DataFrame: The resulting Spark DataFrame.
+        """
+        spark_df = self.session.createDataFrame(df)
+        return spark_df
+
+    def _spark_df_to_pandas_df(self, spark_df: spark.DataFrame) -> pd.DataFrame:
+        """
+        Convert a Spark DataFrame to a Pandas DataFrame.
+
+        Args:
+            df (spark.DataFrame): The Spark DataFrame to be converted to a Pandas DataFrame.
+
+        Example:
+        ```python
+        from viadot.sources import Databricks
+        databricks = Databricks()
+        list = [{"id":"1", "name":"Joe"}]
+        df = pd.DataFrame(list)
+        spark_df = databricks._pandas_df_to_spark_df(df)
+        pandas_df = databricks._spark_df_to_pandas_df(spark_df)
+        ```
+        Returns:
+            pd.DataFrame: The resulting Pandas DataFrame.
+        """
+        return spark_df.toPandas()
+
+    def run(
+        self, query: str, fetch_type: Literal["spark", "pandas"] = "spark"
+    ) -> Union[spark.DataFrame, pd.DataFrame, bool]:
+        """
+        Execute an SQL query.
+
+        Args:
+            query (str): The query to execute.
+            fetch_type (Literal, optional): How to return the data: either
+            in the default Spark DataFrame format or as a Pandas DataFrame. Defaults to "spark".
+
+        Example:
+        ```python
+        from viadot.sources import Databricks
+        databricks = Databricks()
+        query_result = databricks.run("SELECT * FROM schema.table_1")
+        ```
+        Returns:
+            Union[spark.DataFrame, pd.DataFrame, bool]: Either the result set of a query or,
+            in case of DDL/DML queries, a boolean describing whether
+            the query was excuted successfuly.
+        """
+        allowed_fetch_type_values = ["spark", "pandas"]
+        if fetch_type not in allowed_fetch_type_values:
+            raise ValueError(
+                f"Only the values {allowed_fetch_type_values} are allowed for 'fetch_type'"
+            )
+
+        query_clean = query.upper().strip()
+        query_keywords = ["SELECT", "SHOW", "PRAGMA", "DESCRIBE"]
+
+        query_result = self.session.sql(query)
+
+        if any(query_clean.startswith(word) for word in query_keywords):
+            if fetch_type == "record":
```

Comment:
Correct me if I'm wrong but possible values for `fetch_type` are `spark` and `pandas` as far as I can see, so this statement would never execute.